src/Server/Main.cpp

```diff
@@ -65,6 +66,12 @@ ConfigPtr Initialize(const std::optional<fs::path>& config_path, const Environme
 	try
 	{
 		pConfig = Config::Load(config_path, environment);
+		if(config_path.has_value()){
+			pConfig->SetConfigPath(config_path.value());
```

Comment:
could `src\Core\Global.cpp`  be a better place? let's not break the pattern.

Reply:
Yep, 2 options come to mind:
1. Store it as a new field in global with a getter/setter
2. Wrap ConfigPtr with a new class (ConfigFile? LoadedConfig?) that has 2 fields: path & values(the ConfigPtr). Then store that in global instead of the ConfigPtr.